### DISCUSSION TITLE
Add platform lift move after print done

### DISCUSSION THREAD

**Discussion by d-kozminihs** [sn4k3/UVtools#247] [Ideas]
I would like to make my Photon Mono X lift platform to the max height after print is done (just like most other printers do). It would be cool if user could set a certain safe level (from 0) where platform would stop.

**Comment by sn4k3** [comment]
Bad firmwares... Every firmware should lift the end print to a level...
This is possible via scripting, i guess if you append a layer at Z=240 it will work

**Reply by d-kozminihs** [reply]
I agree, but Anycubic does not seem to be willing to introduce this lifting (maybe at some point in future, but latest releases didn't help). I will have to look into scripting, not even sure how it works...

**Reply by sn4k3** [reply]
I can do one script and publish for you. Still im away, can only do it in about 8 days

**Reply by d-kozminihs** [reply]
That would be very much appreciated (not only by me). I have looked into scripting, I doubt I could learn it in less than 8 days anyway :D There is a community on Facebook (Anycubic Photon Mono X) where members would love to see your script, if you don't mind

**Comment by sn4k3** [comment]
![UVtools_2021-07-31_20-30-48](https://user-images.githubusercontent.com/113281/127750587-3c234f00-2f47-49f5-b2d3-91609026e113.png)

**Reply by d-kozminihs** [reply]
Hi sn4k3, thanks a lot, is it published already? Can't find it to test. 
P.S.: there's a typo in last sentence: "Not compatible WITH all printers..."

**Reply by sn4k3** [reply]
Will release latter.
Will fix the typo

**Reply by sn4k3** [reply]
It's released, test and report back

**Reply by d-kozminihs** [reply]
Unfortunately, Mono X ignores this change. It "prints" that extra layer at the same thickness as all other layers. I am assuming it reads layer thickness once in the beginning and sets some firmware sequence accordingly... I wish it could understand regular Gcode... G1 Z240 :D

**Reply by sn4k3** [reply]
> Unfortunately, Mono X ignores this change. It "prints" that extra layer at the same thickness as all other layers. I am assuming it reads layer thickness once in the beginning and sets some firmware sequence accordingly... I wish it could understand regular Gcode... G1 Z240 :D

That's why i would never buy no gcoded printers. My first and only buy is an IBEE, total freedom :)
BUT mono X is capable of diferent layer heights, are you on latest firmware? Also is possible that the layer diference is that much that firmware "auto fix", OR maybe you can try a lower Z, like 200?

If the diference is too much from last layer to this layer, a fix could be add a sequential amount of layers in *mm increments, but that would be slow to perform...

**Reply by d-kozminihs** [reply]
"You get what you pay for" - I paid £310 for Mono X, no wonder it is somewhat problematic :)
I am using the latest firmware. First, I was afraid to use 245mm, so I set it to 50mm, then 100mm, then 200mm... No difference. 
I have no other suggestions for now. Will see if Anycubic would be so kind to fix that... one day :D

**Reply by sn4k3** [reply]
They just don't care to deliver better quality. All firmware issues could be fixed without much effort but...

**Reply by d-kozminihs** [reply]
Thank you for your effort :)

**Comment by printfoam** [comment]
I'd like to have a raise and pause portion of the print...
for example:


layer 1: z = 0.05
layer 2: z = 0.10
layer 3: z = 150.00 (pause layer: raise build 150 mm up, and then do a all black image for exposure time...)
layer 4: z = 0.15 (resume printing) 

I tried modifing a pwmx file with UVTools with this sort of scenario but it did not work.
I have also tried setting bottom layers to 0 as per @helgekeck suggestion. 
As well as trying small raise heights, and using layer 2 retract height as a proxy to get a larger lift in between layers. 
None of these attempts worked with firmware version 0.2.3

**Reply by sn4k3** [reply]
Your best shot is to use the Timelapse Tool as following in the desired layer to raise:
![image](https://github.com/sn4k3/UVtools/assets/113281/a689964d-0fa1-49ee-8dde-634b7fa8334a)

However Anycubic is very limited and can't wait on a position, so you need to set a very slow speed. Still i doubt it will work at all

**Reply by printfoam** [reply]
@sn4k3 
do you expect this to work with the 0.2.2 or 0.2.3 firmware?

**Reply by sn4k3** [reply]
I don't have any Anycubic machine, but I won't expect it to work at all in any near future for the current machines/format

**Reply by printfoam** [reply]
@sn4k3  - are you aware of a motherboard replacement for the mono? was there a printer with gcode that you recommend?

**Reply by sn4k3** [reply]
I'm not aware of any mod you can fit into the printer.
gcode msla printers are rare, so far the most recent and bigger gcode printer i know of is the [Bene6](https://www.nova3dp.com/products/nova3d-bene6-resin-3d-printer-with-6-6-4k-lcd) Still I don't know about the machine quality. Format is full capable and gcoded but hardware quality is unknown by me.
I have uniz ibee which is also gcode have its things but once you know them it's almost bullet proof, good hardware, lame software/firmware.
[Nova Whale3](https://www.nova3dp.com/products/whale3-super) also new and 8K with big screen, but I can't confirm yet if is gcode or CTB
